### ISSUE TITLE
Is `eth_utils.abi.collapse_if_tuple` private or public API?

### ISSUE THREAD

**Issue by davesque** [ethereum/eth-utils#144]
_Originally posted by @pipermerriam in https://github.com/ethereum/eth-utils/pull/143#issuecomment-448722217_

**Comment by carver** [comment]
Right, so I think `collapse_if_tuple()` belongs in `eth-abi`, except that `eth-abi` depends on `eth-utils`, and `eth-utils` uses the method for `function_abi_to_4byte_selector()` and `event_abi_to_log_topic()`.

I think maybe the *right* solution is to deprecate those last two methods, with the intention to move them into `eth-abi`. Then `collapse_if_tuple` can move into `eth-abi` too.

So maybe `_` prefix is best to indicate this weird scenario, that `collapse_if_tuple()` will be leaving `eth-abi` altogether. (assuming this is the plan we coalesce on)
